nemo/collections/nlp/data/language_modeling/megatron/gpt_sft_chat_dataset.py

```diff
@@ -29,18 +29,31 @@
 SYSTEM_TOKEN = "<extra_id_0>System\n"
 TURN_TOKEN = "<extra_id_1>"
 
-GUARD_RAIL_INSTRUCTION = {
+TYPE_INSTRUCTION = {
     "TEXT_TO_CANONICAL_FORM": "Given a dialogue, for each turn you need to generate a short summary called a canonical form. Generate the canonical form for the last turn in the dialogue.",
     "CANONICAL_FORM_TO_TEXT": "Given a dialogue, for each turn we also have a short summary called a canonical form. Generate the canonical form given the last turn message and canonical form. Then generate the message.",
+    'TEXT_TO_VALUE': "",
+    'VALUE_TO_TEXT': '',
 }
 
 
-def _mask_targets(target, tokenized_lens, speakers, header_len, s_ids, tokenizer, mask_role):
+def _mask_targets(target, tokenized_lens, speakers, header_len, s_ids, tokenizer, mask_role, gtype, extra_id_2_id):
```

Comment:
Can you write a description on what the code is doing using an example such as (also include the control tokens)

e.g. 

(user utterance1)  (bot utterance1 )  (user utterance2) (bot utterance)
XXXXXXXXXX       YYYY                     ZZZZZZZZZZZZ   AAAA → loss on YYYY and AAAA together

as well as what each arg does?

This function is pretty hard to read without it.

Reply:
added doc